Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1528,6 +1528,17 @@ public InputActionReference trackedDevicePosition
             set => SwapAction(ref m_TrackedDevicePositionAction, value, m_ActionsHooked, m_OnTrackedDevicePositionDelegate);
         }
 
+        // We should dispose the static default actions thing because otherwise it will survive domain reloads
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void ResetDefaultActions()
+        {
+            if (defaultActions != null)
+            {
+                defaultActions.Dispose();
```

Comment:
I guess, calling Dispose is just to stay on the safe side since the only data default actions stores is a ref to ScriptableObject that won't survive even with domain reload off. It will be in this intermediate state though, where its native data is gone but the instance is not null until the next gc run (however you'll see it as "null" in debugger, and ==null will be true as well, thanks to Unity's Equals overload for all Object descendants).

Reply:
Does this comment imply that it should be `Object.ReferenceEquals(null, defaultActions)` in if statement above? If its disposable, yes it should be disposed, even if the implementation would do nothing IMO.